### PR TITLE
🚀 Upgrade to .Net 6.0

### PR DESCRIPTION
**Pull request by PureKrome**
YAY!

Also includes some specific stuff like the new c# 8/9/10 features, like nullable enabled, etc.

### CODE REVIEW

**Comment by PureKrome** [comment]
sigh. tests busted.

**Comment by PureKrome** [comment]
Good to go, again.

**Comment by RPM1984** [comment]
Do we know why coverage has dropped 11%?

**Comment by PureKrome** [comment]
Yes - found it hard to test some extensions helper class

File: src/Homely.AspNetCore.Mvc.Helpers/Models/OpenApiSettings.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        public string Title { get; init; } = DefaultOpenApiTitle;
+        public string Version { get; init; } = DefaultOpenApiVersion;
+        public string RoutePrefix { get; init; } = DefaultOpenApiRoutePrefex;
+    }
+}
+
+// Below is need for records to work on netcoreapp3.1
+namespace System.Runtime.CompilerServices
+{
+    public class IsExternalInit { }
```
Comment (RPM1984) [inline src/Homely.AspNetCore.Mvc.Helpers/Models/OpenApiSettings.cs]: is this still req'd?

File: tests/TestWebApplication/TestWebApplication.csproj [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
-    <LangVersion>latest</LangVersion>
+    <TargetFramework>net6.0</TargetFramework>
   </PropertyGroup>
 
-  <ItemGroup>
-    <PackageReference Include="Hellang.Middleware.ProblemDetails" Version="5.1.1" />
-  </ItemGroup>
+  <!--<ItemGroup>
+    <PackageReference Include="Hellang.Middleware.ProblemDetails" Version="5.3.0" />
+  </ItemGroup>-->
```
Comment (RPM1984) [inline tests/TestWebApplication/TestWebApplication.csproj]: pls remove

**Review by RPM1984 — CHANGES_REQUESTED** [review]

File: src/Homely.AspNetCore.Mvc.Helpers/Models/OpenApiSettings.cs (hunk 1 above)
Comment (PureKrome) [reply src/Homely.AspNetCore.Mvc.Helpers/Models/OpenApiSettings.cs]: 🔧

File: tests/TestWebApplication/TestWebApplication.csproj (hunk 2 above)
Comment (PureKrome) [reply tests/TestWebApplication/TestWebApplication.csproj]: 🔧

**Review by RPM1984 — APPROVED** [review]

**Pull request merged**